ibc-core/ics02-client/context/src/consensus_state.rs

```diff
@@ -16,5 +17,5 @@ pub trait ConsensusState: Send + Sync + Convertible<Any> {
     fn root(&self) -> &CommitmentRoot;
```

Comment:
Can we also change `root()` to return to just `CommitmentRoot`? Actually, this should be `Cow<'_, CommitmentRoot>` but `Cow` is only available on `std`.

Reply:
I wonder if you've come across any limitations because of this in any of the integration projects we've worked on? I'm trying to get a better sense of whether it's really necessary.

Reply:
Currently, we don't have any limitations - except in the `MockConsensusState` - as our Tendermint Light Client's consensus state just stores the root as `CommitmentRoot` type.

But the issue is, we can't assume how the implementor of `ConsensusStateTrait` is storing the commitment root. `fn root(&self)` can only return a reference of `CommitmentRoot` if the commitment root is stored as `CommitmentRoot` in `Self`.

For example, the following `DummyConsensusState` can't implement the trait.

```rs
struct DummyConsensusState{ root: Vec<u8> };
```

Reply:
Got it, I see now what you're saying. Right, it implicitly forces developers to use `CommitmentRoot` as the `root` type. However, one concern I see with this update is that everywhere we call `.root()` we'll end up with unnecessary cloning and increased overhead. Alternatively, we could make `fn root(self)` take ownership of self, but that would still require cloning the consensus state in multiple places since we need `timestamp()` too from the consensus state in those instances. I'm actually prioritizing efficiency over flexibility here.
I suggest we hold off on this change until we get a clear request from our clients. Further improvements here looks to me be pushing us towards using associated types or generics, which would need more thoughts. So, it’s probably better to handle this in a separate PR since it’s outside the scope of this one. What do you think?

Reply:
> we'll end up with unnecessary cloning

Yea [`Cow`](https://doc.rust-lang.org/stable/std/borrow/enum.Cow.html) is the solution for that :smile: But it requires `std`.

Alright, let's fix this in future.